lib/active_merchant/billing/gateways/usa_epay_advanced.rb

```diff
@@ -348,12 +348,71 @@ def add_customer(options={})
       #  * Same as add_customer
       #
       def update_customer(options={})
+        ActiveMerchant.deprecated "update_customer is deprecated and will be removed from a future release of ActiveMerchant. Please use update instead."
         requires! options, :customer_number
 
         request = build_request(__method__, options)
         commit(__method__, request)
       end
 
+      # Update a customer by replacing either all of the customer details or changing only a few fields
+      #
+      #
+      # ==== Required
+      # * <tt>:customer_number</tt> -- customer to update
+      #
+      # 
+      # To update only a few fields use this set of options
+      #
+      # ==== Options
+      # * <tt>:fields</tt> -- array of 2 element long arrays representing FieldValue object
+      #   * <tt>:Field</tt> -- name of the field
+      #   * <tt>:Value</tt> -- field value 
+      #
+      # To replace customer details use this set of options
+      #
+      # ==== Options
+      # * <tt>:id</tt> -- merchant assigned id
+      # * <tt>:notes</tt> -- notes about customer
+      # * <tt>:data</tt> -- base64 data about customer
+      # * <tt>:url</tt> -- customer website
+      # * <tt>:billing_address</tt> -- usual options
+      # * <tt>:payment_methods</tt> -- array of payment method hashes.
+      #   * <tt>:method</tt> -- credit_card or check
+      #   * <tt>:name</tt> -- optional name/label for the method
+      #   * <tt>:sort</tt> -- optional integer value specifying the backup sort order, 0 is default
+      #
+      # ==== Recurring Options
+      # * <tt>:enabled</tt> -- +true+ enables recurring
+      # * <tt>:schedule</tt> -- daily, weekly, bi-weekly (every two weeks), monthly, bi-monthly (every two months), quarterly, bi-annually (every six months), annually, first of month, last day of month
+      # * <tt>:number_left</tt> -- number of payments left; -1 for unlimited
+      # * <tt>:next</tt> -- date of next payment (Date/Time)
+      # * <tt>:amount</tt> -- amount of recurring payment
+      # * <tt>:tax</tt> -- tax portion of amount
+      # * <tt>:currency</tt> -- numeric currency code
+      # * <tt>:description</tt> -- description of transaction
+      # * <tt>:order_id</tt> -- transaction order id
+      # * <tt>:user</tt> -- merchant username assigned to transaction
+      # * <tt>:source</tt> -- name of source key assigned to billing
+      # * <tt>:send_receipt</tt> -- +true+ to send client a receipt
+      # * <tt>:receipt_note</tt> -- leave a note on the receipt
+      #
+      # ==== Point of Sale Options
+      # * <tt>:price_tier</tt> -- name of customer price tier
+      # * <tt>:tax_class</tt> -- tax class
+      # * <tt>:lookup_code</tt> -- lookup code from customer/member id card; barcode or magnetic stripe; can be assigned by merchant; defaults to system assigned if blank
+      #
+      # ==== Response
+      # * <tt>#message</tt> -- customer number assigned by gateway
+
+      def update(options={})
```

Comment:
Per AM's standard API, this should be `update(identifier, options={})`. I'm guessing `identifier` would be `customer_number`?